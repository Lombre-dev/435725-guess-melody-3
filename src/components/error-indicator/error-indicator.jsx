import PropTypes from 'prop-types';
import React from 'react';
import {connect} from 'react-redux';

function ErrorIndicator({count, limit}) {

  const iterator = new Array(limit).fill(``);

  return (
    <div className="game__mistakes">
      {
        iterator.map((_value, index) => {
          return (
            <div key={`error-${index}`} className="wrong" style={{opacity: index < count ? 0.25 : 1}} />
          );
        })
      }
    </div>
  );
}

ErrorIndicator.propTypes = {
  count: PropTypes.number.isRequired, // from store
  limit: PropTypes.number.isRequired, // from store
};

function mapStateToProps(state) {
  return {
    count: state.errorsCount,
    limit: state.errorsLimit,
  };
}

export {ErrorIndicator};
export default connect(mapStateToProps)(ErrorIndicator);
