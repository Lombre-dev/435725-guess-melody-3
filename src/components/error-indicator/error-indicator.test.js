import React from 'react';
import renderer from 'react-test-renderer';
import {ErrorIndicator} from './error-indicator';

const ERRORS_COUNT = 0;
const ERRORS_LIMIT = 3;

describe(`<ErrorsIndicator />`, () => {

  it(`render should be match markup`, () => {

    const result = renderer
      .create(<ErrorIndicator
        count={ERRORS_COUNT}
        limit={ERRORS_LIMIT}
      />)
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
