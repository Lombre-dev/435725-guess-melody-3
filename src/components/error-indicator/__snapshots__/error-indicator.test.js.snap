// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorsIndicator /> render should be match markup 1`] = `
<div
  className="game__mistakes"
>
  <div
    className="wrong"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <div
    className="wrong"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <div
    className="wrong"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
</div>
`;
